src/transport_ofi.c

```diff
@@ -980,6 +981,38 @@ int query_for_fabric(struct fabric_info *info)
     return ret;
 }
 
+static int shmem_transport_ofi_target_ep_init() {
+    int ret = 0;
+
+    struct fabric_info* info = &shmem_transport_ofi_info;
+    info->p_info->ep_attr->tx_ctx_cnt = FI_SHARED_CONTEXT;
+    info->p_info->caps = FI_RMA | FI_ATOMICS;
+    info->p_info->tx_attr->op_flags = FI_DELIVERY_COMPLETE;
+    info->p_info->mode = 0;
+    info->p_info->tx_attr->mode = 0;
+    info->p_info->rx_attr->mode = 0;
+
+    ret = fi_endpoint(shmem_transport_ofi_domainfd,
+                      info->p_info, &shmem_transport_ofi_target_ep, NULL);
+    OFI_CHECK_RETURN_MSG(ret, "target endpoint creation failed (%s)\n", fi_strerror(errno));
+
+    /* Attach the shared context */
+    ret = fi_ep_bind(shmem_transport_ofi_target_ep, &shmem_transport_ofi_stx->fid, 0);
+    OFI_CHECK_RETURN_STR(ret, "fi_ep_bind STX to target endpoint failed");
+
+    /* Attach the address vector */
+    ret = fi_ep_bind(shmem_transport_ofi_target_ep, &shmem_transport_ofi_avfd->fid, 0);
```

Comment:
I think this is also unnecessary, since the EP is not used to transmit.

Reply:
When I remove this, I get a "bad access" seg fault in the RX progress code:
``` 
  frame #0: 0x00000001002959fe libfabric.1.dylib`sock_av_get_addr_index(av=0x0000000000000000, addr=0x0000000101f00000) at sock_av.c:71
    frame #1: 0x00000001002b9528 libfabric.1.dylib`sock_pe_process_rx_conn_msg(pe=0x000000010035b000, rx_ctx=0x0000000100603920, pe_entry=0x00000001003841a0) at sock_progress.c:1474
    frame #2: 0x00000001002b69d1 libfabric.1.dylib`sock_pe_process_recv(pe=0x000000010035b000, rx_ctx=0x0000000100603920, pe_entry=0x00000001003841a0) at sock_progress.c:1542
    frame #3: 0x00000001002b30ee libfabric.1.dylib`sock_pe_progress_rx_pe_entry(pe=0x000000010035b000, pe_entry=0x00000001003841a0, rx_ctx=0x0000000100603920) at sock_progress.c:2035
    frame #4: 0x00000001002b32c9 libfabric.1.dylib`sock_pe_progress_rx_ctrl_ctx(pe=0x000000010035b000, rx_ctx=0x0000000100603920, tx_ctx=0x00000001006036f0) at sock_progress.c:2454
    frame #5: 0x00000001002b3458 libfabric.1.dylib`sock_pe_progress_tx_ctx(pe=0x000000010035b000, tx_ctx=0x00000001006036f0) at sock_progress.c:2487
    frame #6: 0x00000001002b4e05 libfabric.1.dylib`sock_pe_progress_thread(data=0x000000010035b000) at sock_progress.c:2648
```
Am I missing something, or do we indeed need to bind the AV?

Reply:
Looks like I'm wrong -- the fi_ep_bind manpage says "Connectionless endpoints must be bound to a single address vector."

It would be sorta nice if fi_ep_enable did a sanity check on the EP...